src/utils/formatSearchResponse.ts

```diff
@@ -54,20 +69,25 @@ const formatSearchResponse = async (
   return finalResponse;
```

Comment:
this return statement executes before the forEach async callback completes. fortunately you don't have any non i/o blocking call inside the async function so this script works.
possible solution is to push all your async function to array and do 
`await Promise.all([])`

Reply:
Yes, correct. Sorry for not noticing this earlier. Now, it has been modified.